### PR TITLE
[Enhancement] Add `menuRenderer` prop to support windowing for large lists

### PR DESCRIPTION
**Pull request by bvaughn**
I believe this PR resolves the following open issues: #554, #674, #739, #771, #807.

This PR adds support for windowing libraries like [react-virtualized](https://github.com/bvaughn/react-virtualized) to be used to render large lists of options. It is similar to the approach mentioned in PR #763 but it avoids creating all of the menu options when rendering and so it performs better.

The size of the PR is mostly due to the fact that it contains a JSON data file of the largest 1,000 cities in Europe (used to demonstrate custom menu windowing). The actual changes to the `Select` component were very minimal (~40 new lines).

#### This change-set includes:
* New `renderMenu` prop added to `Select` component.
* New tests added to `test/Select-test.js` for new property. (Previous tests all pass.)
* README [updated to mention](https://github.com/bvaughn/react-select/#effeciently-rendering-large-lists-with-windowing) newly added property.
* Example page updated to demonstrate new property as well. (Live demo of this functionality can be seen [here, under "World's Largest Cities" header](https://rawgit.com/bvaughn/react-select/demo/examples/dist/index.html).

Note that [react-virtualized](https://github.com/bvaughn/react-virtualized) was added to the `devDependencies` section (not `dependencies`) so it could be used in the example to demonstrate a more efficient way of rendering select menus with large amounts of data.

If this PR gets merged, or one like it, I plan to release a small adapter library with a single HOC that provides the react-virtualized drop-down menu automatically and passes through other props to `Select`. (In other words, I'm happy to take over the ownership of the feature.)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -250,6 +296,7 @@ For multi-select inputs, when providing a custom `filterOptions` method, remembe
 	optionRenderer	|	func	|	undefined	|	function which returns a custom way to render the options in the menu
 	options 	|	array	|	undefined	|	array of options
 	placeholder 	|	string\|node	|	'Select ...'	|	field placeholder, displayed when there's no value
+	renderMenu | func | undefined | Renders a custom menu with options; accepts the following named parameters: `renderMenu({ focusedOption, focusOption, options, selectValue, valueArray })`
```
Comment (cloudkite) [inline README.md]: maybe `menuRenderer` would be better choice since it follows existing naming convention of `optionRenderer `

**Comment by cloudkite** [comment]
Nice work! :+1: 

I think it would be good idea not to include `examples/dist` and `lib` folders in the PR, makes it alot more readable

**Comment by bvaughn** [comment]
Truth. I wanted a live demo to point out so I had to push them to my fork. But maybe I'll clean up this PR and use another branch for the demo. Thanks for the input :)

File: README.md (hunk 1 above)
Comment (bvaughn) [inline README.md]: Renamed. Thanks for the suggestion.

**Comment by bvaughn** [comment]
Okay. Dist files have all been removed. Only large files still around are the example cities list.

**Comment by JedWatson** [comment]
This looks fantastic @bvaughn! Thanks for the PR, and for owning the feature :)

**Comment by Naveg** [comment]
This is great. It seems like it should be possible to use `react-tether` in the `menuRenderer` function, but I'm having trouble getting that to work. Problem is that I think the tether target should be the `Select-control` component, which isn't available here. Only thing I can think of it to add a dependency on `react-tether` and wrap the `Select-control` and menu in a `TetherComponent`, which we definitely don't want. Any ideas?

**Comment by Craga89** [comment]
:+1: to this. This is way more flexible than the `menuComponent` PR I made, great job @bvaughn! I'd love to see this merged so we can translate our current implementation over!

**Pull request merged**